### PR TITLE
feat(types): update types to use alloy

### PR DESCRIPTION
**Pull request by dancoombs**
## Proposed Changes

  - Downgrade alloy due to https://github.com/alloy-rs/core/issues/717
  - Remove contracts from types crate
  - Update types crate to use alloy and new rundler-contracts crate

### CODE REVIEW

File: crates/contracts/src/v0_6.rs [hunk 1]
```diff
@@ -40,10 +40,44 @@ sol! {
         bytes signature;
     }
 
+    #[allow(missing_docs)]
+    #[derive(Default, Debug, PartialEq, Eq)]
+    struct ReturnInfo {
+        uint256 preOpGas;
```
Comment (0xfourzerofour) [inline crates/contracts/src/v0_6.rs:46]: is this valid rust?

Comment (0xfourzerofour) [reply crates/contracts/src/v0_6.rs:46]: ok its wrapped in sol soz

File: crates/types/src/user_operation/v0_6.rs [hunk 2]
```diff
@@ -11,25 +11,22 @@
...
-pub use crate::contracts::v0_6::i_entry_point::{UserOperation, UserOpsPerAggregator};
 use crate::{
     chain::ChainSpec,
     entity::{Entity, EntityType},
     EntryPointVersion,
 };
 
 /// Gas overhead required by the entry point contract for the inner call
-pub const ENTRY_POINT_INNER_GAS_OVERHEAD: U256 = U256([5_000, 0, 0, 0]);
+pub const ENTRY_POINT_INNER_GAS_OVERHEAD: U256 = uint!(5000_U256);
```
Comment (0xfourzerofour) [inline crates/types/src/user_operation/v0_6.rs:29]: this macro is so much better

File: crates/contracts/src/v0_6.rs (hunk 1 above)
Comment (dancoombs) [reply crates/contracts/src/v0_6.rs:46]: Yeah tbh this is really weird to see, Rust proc macros on solidity code. Powerful though

**Pull request closed without merging**